### PR TITLE
Adds mvp functionality

### PR DESCRIPTION
**Pull request by looselycoupled**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	-rm -rf $(PROJECT).egg-info
+	-rm -rf .eggs
+	-rm -rf site
+	-rm -rf docs/_build
+	-rm -rf platform-builds meta.yaml
+
+# Generate new grpc code
+grpc:
+	@echo Generating files:
+	python -m grpc_tools.protoc -I eventproc/protobuff --python_out=eventproc/protobuff --grpc_python_out=eventproc/protobuff eventproc/protobuff/api.proto
```
Comment (oridb) [inline Makefile:31]: Minor nit: usually it's just one f.

File: eventproc/eventproc.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+def register(name, hook, apikey, notify_on_success, notify_on_failure, flags=[]):
+    """
+    decorator to submit (register) an event handler function
+    """
+
+    # inner will actually receive the decorated func but we still have access
+    # to the args & kwargs due to closure/scope.
+    def inner(func):
+
+        # call grpc service to register event handler
```
Comment (oridb) [inline eventproc/eventproc.py:88]: Should this be uploading the function to s3 as well? Is the plan to do this as a separate PR?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    'Programming Language :: Python :: 3.6',
+    'Programming Language :: Python :: 3.7',
+    'Topic :: Database',
+    'Topic :: Software Development :: Libraries :: Python Modules',
+)
+
+## Important Paths
+PROJECT      = os.path.abspath(os.path.dirname(__file__))
+REQUIRE_PATH = "requirements.txt"
+VERSION_PATH = os.path.join(PACKAGE, "version.py")
```
Comment (oridb) [inline setup.py:64]: This file seems to be missing.

File: eventproc/eventproc.py (hunk 2 above)
Comment (looselycoupled) [reply eventproc/eventproc.py:88]: The client library shouldnt be trying to upload to s3.  At the least that would mean the users would have access to AWS keys.  Uploading is handled by the go service.

File: setup.py (hunk 3 above)
Comment (looselycoupled) [reply setup.py:64]: agreed.  I havent tried a pip install yet

File: Makefile (hunk 1 above)
Comment (looselycoupled) [reply Makefile:31]: Ugh, you're absolutely right.  I'm gonna leave as is but change in a future PR if you dont mind.

File: eventproc/eventproc.py (hunk 2 above)
Comment (oridb) [reply eventproc/eventproc.py:88]: Right, but I'm not seeing where we actually serialize the function and send it to the go service. Maybe I'm just missing something.

Comment (looselycoupled) [reply eventproc/eventproc.py:88]: ahhh.. yes I'm an idiot.  The service has been putting zip files into s3 so I assumed I was sending the function along to it.  Will fix in next commit

**Review by oridb — APPROVED** [review]

**Comment by looselycoupled** [comment]
Ready for final PR again.  This code IS NOT PERFECT but is good enough for an MVP.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# GRPC / Protobuff related
+grpcio>=1.16.1
+grpcio-tools>=1.16.1
```
Comment (oridb) [inline requirements.txt:3]: should this have dill?

File: btrdbextras/eventproc/protobuff/api_pb2_grpc.py.bak [hunk 5]
```diff
@@ -0,0 +1,165 @@
+# Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
```
Comment (oridb) [inline btrdbextras/eventproc/protobuff/api_pb2_grpc.py.bak:1]: Do we want to add the .bak file?

File: btrdbextras/eventproc/eventproc.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    name: str
+        Friendly name of this event handler for display purposes.
+    hook: str
+        Name of the hook that this event handler responds to.
+    notify_on_success: str
+        Email address of user to notify when event handler completes successfully.
+    notify_on_failure: str
+        Email address of user to notify when event handler does not complete
+        successfully.
+    dependencies: str
```
Comment (oridb) [inline btrdbextras/eventproc/eventproc.py:193]: I'd remove this for now, since we don't have support for doing anything with this yet. It's a kwarg, so we can add it back later without a breaking cahnge.

File: docs/source/eventproc.rst [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+Notifications
+~~~~~~~~~~~~~~~~~~~
+
+After your event handler is executed, an email is sent to the success or failure email address supplied during registration.  If successful, the email will contain any text that was returned by your Python function.  If an exception is raised, then the handler code is regarded as unsuccesful and a notification is sent to the corresponding failure email address with exception details.
+
+Dependencies
+~~~~~~~~~~~~~~~~~~~
+
+Our platform will provide a base python 3.7.3 environment with which you can add any desired libraries installable with `pip`.  When registering an event handler you can provide the content for a requirements file as if you had run `pip freeze` at the command line.  We provide some standard libraries such as:
```
Comment (oridb) [inline docs/source/eventproc.rst:17]: Again, as far as pip additions, I thought we were punting to v2.

File: docs/source/eventproc.rst [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+Flags
+~~~~~~~~~~~~~~~~~~~
+
+Each event handler also includes a `flags` argument.  This allows the user to choose which event handlers should run in any given scenarion.  For instance, if submitting new COMTRADE files to the ingress, you can choose the "voltage-transform" flag to indicate that any event handlers registered with this flag will be executed.  Multiple flags can be added to your event handler, and users can choose from multiple flags to determine which handlers should run.  In the latter case, a logical OR is used such that any handler with at least one of the chosen flags will be scheduled for execution.
+
+Connections
+-----------
+
+Each of the following calls require a `Connection` object (different from the btrdb library connection object) as the first argument.  When registering a new event handler, the API key in the `Connection` object will be used as the execution time security context - meaning all code will be executed as the registering user.
```
Comment (oridb) [inline docs/source/eventproc.rst:39]: Just a nit -- to avoid confusing developers, is there any other name we can use? EventContext, maybe?

File: docs/source/eventproc.rst [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+        notify_on_success: str
+            Email address of user to notify when event handler completes successfully.
+        notify_on_failure: str
+            Email address of user to notify when event handler does not complete
+            successfully.
+        dependencies: str
+            A string containing required libraries and versions.  Format should follow
+            a standard `pip freeze` or requirements.txt file.  The process running your
+            handler will `pip install -r` this content.
+        flags: list of str
```
Comment (oridb) [inline docs/source/eventproc.rst:110]: Let's make the empty list behavior explicit here. "An empty list will match all flags."

File: requirements.txt (hunk 4 above)
Comment (looselycoupled) [reply requirements.txt:3]: ugh, yes

File: btrdbextras/eventproc/protobuff/api_pb2_grpc.py.bak (hunk 5 above)
Comment (looselycoupled) [reply btrdbextras/eventproc/protobuff/api_pb2_grpc.py.bak:1]: you mean the .bak file referenced in Makefile?  I would not want to include that particular file but I feel like you are referring to something else?

File: btrdbextras/eventproc/eventproc.py (hunk 6 above)
Comment (looselycoupled) [reply btrdbextras/eventproc/eventproc.py:193]: I've removed from func signature and docs but left in a placeholder.

File: docs/source/eventproc.rst (hunk 7 above)
Comment (looselycoupled) [reply docs/source/eventproc.rst:17]: I must have been confused by this as I put in the code and docs.  I'm gonna remove it in favor of v2.

File: docs/source/eventproc.rst (hunk 8 above)
Comment (looselycoupled) [reply docs/source/eventproc.rst:39]: Hmm.. I'm honestly not sure how I feel about this.  I felt `Connection` was a natural name but don't want to create confusion around it if you think it will do so..  

I'm going to leave in for now partly  because there is a massive replace required for this and I'd like to avoid that testing due to time.

File: docs/source/eventproc.rst (hunk 9 above)
Comment (looselycoupled) [reply docs/source/eventproc.rst:110]: done!

File: btrdbextras/eventproc/protobuff/api_pb2_grpc.py.bak (hunk 5 above)
Comment (looselycoupled) [reply btrdbextras/eventproc/protobuff/api_pb2_grpc.py.bak:1]: ack, sorry must have been looking at something else
removing this file!

**Review by oridb — APPROVED** [review]
